### PR TITLE
Add spatial functions, sort out display of 'Arguments' for all functions, and sundry additions

### PR DESCRIPTION
**Pull request by petraselmer**
This must be forward merged. However, bear in mind that doing so requires the **chunking** work to be done as part of the forward merge.

### CODE REVIEW

**Comment by petraselmer** [comment]
@jjaderberg this adds a new section to the Cypher Refcard. I am unsure as to how this changes it. Let me know if the result looks awkward.

File: manual/cypher/cypher-docs/src/docs/dev/ql/functions/index.asciidoc [hunk 1]
```diff
@@ -138,8 +155,20 @@ These functions are used to manipulate strings or to create a string representat
...
+[[header-query-functions-spatial]]
+**<<query-functions-spatial, Spatial functions>>**
+
+These functions are used to specify points in a 2D coordinate system and to calculate the geodesic distance between two points.
+
+[options="header"]
+|===
+|Function                               |   Description
+|<<functions-point,point()>>            | Returns a point object, given two coordinate values.
+|<<functions-distance,distance()>>      | Returns a floating point number representing in meters the geodesic distance between any two given points.
```
Comment (craigtaverner) [inline manual/cypher/cypher-docs/src/docs/dev/ql/functions/index.asciidoc]: Is 'geodesic' required? Is it correct for cartesian points?

File: manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      """CREATE (copenhagen:TrainStation {longitude: 12.564590, latitude: 55.672874, city: 'Copenhagen'}),
+        |       (malmo:Office {longitude: 12.994341, latitude: 55.611784, city: 'Malmo'}),
+        |
+        |       (copenhagen)-[:TRAVEL_ROUTE]->(malmo)"""
+    )
+    synopsis(
+      "These functions are used to specify points in a 2D coordinate system and to calculate the geodesic distance between two points.")
+    p("Two coordinate reference systems (CRS) are supported: (i) http://spatialreference.org/ref/epsg/4326/[WGS 84] and (ii) http://spatialreference.org/ref/sr-org/7203/[cartesian 2D].")
+    p(
+      """_WGS 84_ is specified with a map containing coordinate values for either `longitude` and `latitude` (this is the default), or `x` and `y`.
```
Comment (craigtaverner) [inline manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala:41]: Does WGS84 support x and y? I thought we required lat and lon for WGS84.

**Review by craigtaverner — APPROVED** [review]

**Comment by petraselmer** [comment]
I've added the DO NOT MERGE label as I want to extend the remit of this PR to include better formatting for all functions first

**Comment by jjaderberg** [comment]
@petraselmer The build that assembles the refcard has moved downstream so we have to add an include for the new content here produced there also, and then we can see how the layout is affected.

**Pull request merged**